cockatrice/src/soundengine.cpp

```diff
@@ -57,46 +85,157 @@ void SoundEngine::soundEnabledChanged()
         audio->deleteLater();
         audio = 0;
     }
+#else //QT5
+    if (settingsCache->getSoundEnabled()) {
+        qDebug("SoundEngine: enabling sound");
+        enabled = true;
+    }else{
+        qDebug("SoundEngine: disabling sound");
+        enabled = false;
+        for (int i = 0; i < fileNames.size(); ++i) {
+            audioData[fileNames[i]]->stop();
+        }
+    }
+#endif
 }
 
 void SoundEngine::playSound(const QString &fileName)
 {
+#if QT_VERSION < 0x050000 //QT4
     if (!audio)
         return;
     
     audio->stop();
     inputBuffer->close();
-    inputBuffer->setData(audioData[fileName]);
+    inputBuffer->setData(audioCache[fileName]);
     inputBuffer->open(QIODevice::ReadOnly);
-#if QT_VERSION >= 0x050000
-    audio->setVolume(settingsCache->getMasterVolume() / 100.0);
-#endif
     audio->start(inputBuffer);
-    
+#else //QT5
+    if (!audioData.contains(fileName) || !enabled){
+        return;
+    }
+
+    audioData[fileName]->setVolume(settingsCache->getMasterVolume());
+    audioData[fileName]->stop();
+    audioData[fileName]->setPosition(0);
+    audioData[fileName]->play();
+#endif
+}
+void SoundEngine::untapStep(){
```

Comment:
I'm not sure I like the proliferation of methods here that just hide string params. This makes it harder to extend. The step names for example will be known via configuration rather than hardcoding once we start migrating to modular game system support